src/CatalogService/DataAccess/Repositories/CategoryRepository.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Data.Common;
+using CatalogService.BusinessLogic.Entities;
+using CatalogService.Contracts.Interfaces;
+using Dapper;
+
+namespace CatalogService.DataAccess;
+
+public class CategoryRepository : IRepository<Category>
```

Comment:
BaseRepository:
- contain SqlConnection property;
- should create SQL connection with command timeout;
- basic command/query execute;
- open connection, close connection, try/catch exceptions, close by finally;
- implement IDisposable pattern

Reply:
used more simplified approach with [sqlclient-sqlconnection](https://learn.microsoft.com/en-us/dotnet/api/microsoft.data.sqlclient.sqlconnection?view=sqlclient-dotnet-standard-5.1)